src/lib/model.ts

```diff
@@ -61,39 +61,50 @@ export class Course {
 	}
 
 	nextRequiredBy() {
-		if (!this.requiredBy || !this.frequency) {
+		const [last, next] = this._currentRecurrancePeriod()
```

Comment:
`Recurrence` ?

Reply:
Couldn't think of anything better - the last and next date the course occurs again (or recurs).

Reply:
Sorry, was suggesting it might be a misspelling — `a` instead of `e`?